libs/openai-assistant/src/lib/assistant/assistant-files.service.ts

```diff
@@ -3,29 +3,103 @@ import { FileObject } from 'openai/resources';
 import { createReadStream } from 'fs';
 import { AiService } from '../ai';
 import { ConfigService } from '../config';
-
+import { AssistantCreateParams, VectorStore } from 'openai/resources/beta';
+import ToolResources = AssistantCreateParams.ToolResources;
+import { AssistantUpdateParams } from 'openai/src/resources/beta/assistants';
+import {
+  AssistantCodeInterpreter,
+  AssistantFileSearch,
+  AssistantToolResources,
+} from './assistant.model';
 @Injectable()
 export class AssistantFilesService {
   constructor(
     private readonly assistantConfig: ConfigService,
     private readonly aiService: AiService,
   ) {}
 
-  async create(
-    fileNames: string[],
+  async getFiles(
+    attachments: string[] = [],
     fileDir = this.assistantConfig.get().filesDir,
-  ): Promise<string[]> {
+  ): Promise<FileObject[]> {
     const files: FileObject[] = [];
 
-    for (const name of fileNames) {
-      const file = await this.aiService.provider.files.create({
-        file: createReadStream(`${fileDir || ''}/${name}`),
-        purpose: 'assistants',
-      });
+    await Promise.all(
+      attachments.map(async item => {
+        const file = await this.aiService.provider.files.create({
+          file: createReadStream(`${fileDir || ''}/${item}`),
+          purpose: 'assistants',
+        });
+
+        files.push(file);
+      }),
+    );
 
-      files.push(file);
+    return files;
+  }
+
+  async getCodeInterpreterResources(
+    data: AssistantCodeInterpreter,
+    fileDir = this.assistantConfig.get().filesDir,
+  ): Promise<ToolResources.CodeInterpreter> {
+    const files = await this.getFiles(data?.fileNames, fileDir);
+
+    return { file_ids: files.map(({ id }) => id) };
+  }
+
+  async getFileSearchResources(
+    data: AssistantFileSearch,
+    fileDir = this.assistantConfig.get().filesDir,
+  ): Promise<ToolResources.FileSearch> {
+    if (!data) {
+      return { vector_store_ids: [] };
     }
 
-    return files.map(({ id }) => id);
+    const vectorStores: VectorStore[] = [];
+
+    await Promise.all(
+      Object.entries(data).map(async ([name, values]) => {
+        if (!values.length) {
+          return;
+        }
+
+        const files = values.map(item =>
+          createReadStream(`${fileDir || ''}/${item}`),
```

Comment:
This interpolation takes a while to decode, maybe we can save it with a meaningfully named const?
Especially, as it has one more occurance above

Reply:
hmmm, I don't get it. Could you tell me more about your idea? 
`fileDir` - is an optional variable in the assistant configuration.  If we define it then it works as a prefix in the path.